rpc/personal_api.go

```diff
@@ -8,47 +8,50 @@ import (
 	"sync"
 	"time"
 
+	"github.com/spf13/viper"
+
+	"github.com/tendermint/tendermint/libs/log"
+
 	sdkcontext "github.com/cosmos/cosmos-sdk/client/context"
 	"github.com/cosmos/cosmos-sdk/client/flags"
 	"github.com/cosmos/cosmos-sdk/crypto/keys"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/mintkey"
 	sdk "github.com/cosmos/cosmos-sdk/types"
-	emintcrypto "github.com/cosmos/ethermint/crypto"
-	params "github.com/cosmos/ethermint/rpc/args"
-	"github.com/spf13/viper"
-	"github.com/tendermint/tendermint/libs/log"
 
 	"github.com/ethereum/go-ethereum/accounts"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/common/hexutil"
 	"github.com/ethereum/go-ethereum/crypto"
+
+	emintcrypto "github.com/cosmos/ethermint/crypto"
+	params "github.com/cosmos/ethermint/rpc/args"
 )
 
-// PersonalEthAPI is the eth_ prefixed set of APIs in the Web3 JSON-RPC spec.
+// PersonalEthAPI is the personal_ prefixed set of APIs in the Web3 JSON-RPC spec.
 type PersonalEthAPI struct {
 	logger      log.Logger
 	cliCtx      sdkcontext.CLIContext
 	ethAPI      *PublicEthAPI
 	nonceLock   *AddrLocker
-	keys        []emintcrypto.PrivKeySecp256k1
 	keyInfos    []keys.Info
 	keybaseLock sync.Mutex
 }
 
-// NewPersonalEthAPI creates an instance of the public ETH Web3 API.
+// NewPersonalEthAPI creates an instance of the public Personal Eth API.
 func NewPersonalEthAPI(cliCtx sdkcontext.CLIContext, ethAPI *PublicEthAPI, nonceLock *AddrLocker, keys []emintcrypto.PrivKeySecp256k1) *PersonalEthAPI {
 	api := &PersonalEthAPI{
 		logger:    log.NewTMLogger(log.NewSyncWriter(os.Stdout)).With("module", "json-rpc"),
 		cliCtx:    cliCtx,
 		ethAPI:    ethAPI,
 		nonceLock: nonceLock,
-		keys:      keys,
 	}
 
 	infos, err := api.getKeybaseInfo()
 	if err != nil {
 		return api
 	}
 
+	api.ethAPI.keys = append(api.ethAPI.keys, keys...)
```

Comment:
not sure if I should append them or just set them to `api.ethAPI.keys = keys`

Reply:
you can probably remove this since ethAPI is already loaded with `keys`